### DISCUSSION TITLE
Column tooltips in DataTable

### DISCUSSION THREAD

**Discussion by hughess** [evidence-dev/evidence#2595] [Ideas]
### Description

![image](https://github.com/user-attachments/assets/7ca06201-f066-4fef-b2a8-94b9fb76f831)

### What problem would this solve?

Provide definitions/context in table without taking up too much space. Make it clear to end user where to get information about a column

### How should it work?

I think this could be a prop in the `Column` component (maybe `tooltip`), which accepts a string.

E.g.,
```
<Column id=first_30_days_sales tooltip="First 30 days counted from the date of first customer purchase"/>
```

**Comment by kennan-simpliphy** [comment]
Yes that would be great.

**Comment by kennan-simpliphy** [comment]
Could the string be html or markdown so that the tooltip can be formatted?

**Comment by hughess** [comment]
That might be tricky, but two ideas come to mind for markdown formatting.

1. Create a component for the tooltip itself and allow passing content through a slot
    ```html
      <Column id=first_30_days_sales>
         <Tooltip>
            **First 30 days** counted from the date of first customer purchase
         </Tooltip>
      </Column>
   ```

2. Allow passing content through a slot in the Column component
   ```html
       <Column id=first_30_days_sales>
           **First 30 days** counted from the date of first customer purchase
      </Column>
   ```

A bit verbose unfortunately
